src/main/java/dev/dirs/impl/Windows.java

```diff
@@ -0,0 +1,205 @@
+package dev.dirs.impl;
+
+import dev.dirs.Constants;
+
+import java.lang.foreign.AddressLayout;
+import java.lang.foreign.Arena;
+import java.lang.foreign.FunctionDescriptor;
+import java.lang.foreign.GroupLayout;
+import java.lang.foreign.Linker;
+import java.lang.foreign.MemoryLayout;
+import java.lang.foreign.MemorySegment;
+import java.lang.foreign.SymbolLookup;
+import java.lang.foreign.ValueLayout;
+import java.lang.invoke.MethodHandle;
+
+import static dev.dirs.impl.Util.isNullOrEmpty;
+import static dev.dirs.impl.Util.stringLength;
+import static java.lang.foreign.ValueLayout.JAVA_BYTE;
+import static java.lang.foreign.ValueLayout.JAVA_CHAR;
+
+public final class Windows {
+
+	private Windows() {}
+
+	static {
+		if (Constants.operatingSystem == 'w') {
+			System.loadLibrary("ole32");
+			System.loadLibrary("shell32");
+		}
+	}
+
+	private static final SymbolLookup SYMBOL_LOOKUP = SymbolLookup.loaderLookup().or(Linker.nativeLinker().defaultLookup());
+	private static final ValueLayout.OfByte C_CHAR = ValueLayout.JAVA_BYTE;
+	private static final ValueLayout.OfShort C_SHORT = ValueLayout.JAVA_SHORT;
+	private static final AddressLayout C_POINTER = ValueLayout.ADDRESS
+			.withTargetLayout(MemoryLayout.sequenceLayout(java.lang.Long.MAX_VALUE, JAVA_BYTE));
```

Comment:
@brcolow can you expand on `Long.MAX_VALUE` here? Wouldn't that create a huge layout?

Reply:
Great question. I actually don't know the answer but it is quite common:

https://github.com/search?q=MemoryLayout.sequenceLayout%28java.lang.Long.MAX_VALUE&type=code

These declarations are actually created by jextract, not me. I just copied them into the file.

Reply:
Clearly it is not doing what it is actually saying as LONG_MAX is 8192 petabytes :). Must be a way to specify unknown size as we don't know how big it is up front?

Reply:
Thanks, good to know. If `jextract` does it it's hopefully fine ... ':-)

Reply:
> By the way, thanks so much for setting me as author - that was very kind of you.

No need to thank, it's the correct thing to do!